kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -302,4 +307,17 @@ private Duration getShutdownDuration() {
         ? config.getDuration(SHUTDOWN_DURATION)
         : Duration.ofSeconds(30);
   }
+
+  private void handleInitialDelay() {
+    try {
+      getLogger()
+          .info(
+              "Sleeping for {} millisecond before kafka streams app is started.",
+              initialDelay.toMillis());
+      TimeUnit.MILLISECONDS.sleep(initialDelay.toMillis());
```

Comment:
nit: skip calling sleep() If `initialDelay` is 0 (default case). Simply return, and isSleeping should continue to be false.

Reply:
TimeUnit.sleep implementation has check for it, so just passed the initial delay to it -
```
       if (timeout > 0) {
            long ms = toMillis(timeout);
            int ns = excessNanos(timeout, ms);
            Thread.sleep(ms, ns);
        }
```

Reply:
Added the check to stop isSleeping boolean state transition